src/main/scala/dpla/ingestion3/mappers/providers/Extractor.scala

```diff
@@ -2,40 +2,88 @@ package dpla.ingestion3.mappers.providers
 
 import java.net.URI
 
-import dpla.ingestion3.model.{DplaMapData, EdmAgent, OreAggregation}
-import dpla.ingestion3.utils.Utils
+import dpla.ingestion3.model.{EdmAgent, OreAggregation}
+import org.apache.commons.codec.digest.DigestUtils
 
-import scala.util.{Failure, Success, Try}
+import scala.util.Try
 
 /**
   * Interface that all provider extractors implement.
   */
 
 trait Extractor {
   // Base item uri
-  private val baseItemUri = "http://dp.la/api/items/"
+  private val baseDplaItemUri = "http://dp.la/api/items/"
 
   def build(): Try[OreAggregation]
   def agent: EdmAgent
+
+  /**
+    * Does the provider use a prefix (typically their provider abbreviation) to
+    * salt the identifier?
+    *
+    * @return Boolean
+    */
+  def useProviderName(): Boolean
+
+  /**
+    * Extract the record's "persistent" identifier. Implementations should raise
+    * an Exception if no ID can be extracted
```

Comment:
Can you add an `@throws` comment here to reflect the expectation an an exception may be thrown?  I don't know if that's a good idea or not, just a thought.

Reply:
we got into trouble passing in options because they weren't getting decomposed and we ended up hashing Some(”id") rather than "id".  When choosing this method I was trying to be as explicit as possible all the way through

I'll take another look at it though to see if there might be another way...

Reply:
Done.